projecttacoma/fqm-testify
# Date pickers

# Summary
Added measurement period date pickers and autofill functionality on measure upload

## New Behavior
- Interactive date pickers now available on home screen
- On measure bundle upload, measurementPeriod date pickers will autofill to the `effectivePeriod` of the measure resource. If effective period is not defined, the period will default to `2022-01-01 - 2022-12-31`, If only the start or end of the effectivePeriod is defined, the datePickers will be populated with a 1 year period based on the value of the defined point.

## Code Changes
- Added DateSelectors component
- Added populateMeasurePeriod function which gets called on measureBundle upload
- Added testing
- Added luxon, dayjs, and @mantine/dates as dependencies

# Testing Guidance
- Run `npm test`
- Start up the app with `npm run dev`
- Upload the EXM130 bundle and ensure the date pickers auto fill to `January, 1 2018` and `December 31, 2018`
- Remove the `effectivePeriod` attribute on the EXM130 measure, then re-upload. Ensure the datePickers update to `January 1, 2022` and `December 31, 2022`
- Add back just the end date, and ensure the date pickers populate to `December 31, 2017` `December 31, 2018`
- Add back just the start date, and ensure the date pickers populate to `January 1, 2018` `January 1, 2019`
- Select dates for yourself and ensure they overwrite the auto-poulated dates
